crates/client/rpc/src/lib.rs

```diff
@@ -180,8 +179,10 @@ where
 
         let substrate_block_hash = self.substrate_block_hash_from_starknet_block(block_id)?;
 
-        let starknet_block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash)
-            .ok_or("Failed to retrieve the substrate block number".to_string())?;
+        let starknet_block = match get_block_by_block_hash(self.client.as_ref(), substrate_block_hash) {
+            Ok(block) => block,
+            Err(err) => return Err(format!("Error getting Starknet block by block hash: {}", err)),
```

Comment:
StarknetRpcApiError::BlockNotFound

Reply:
or just use `?`

Reply:
@tdelabro 
```rust
fn substrate_block_number_from_starknet_block(&self, block_id: BlockId) -> Result<u64, String> {
```
Calling function `substrate_block_number_from_starknet_block` signature requires `String`, Should I return `err.to_string()`?  Thanks

Reply:
Change the funtion signature so that the error type is not `String` anymore but `StarknetRpcApiError`